model/stats/query.go

```diff
@@ -0,0 +1,284 @@
+package stats
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+const (
+	MaxQueryLimit             = 1000
+	GroupByTest       GroupBy = "test"
+	GroupByTask       GroupBy = "task"
+	GroupByVariant    GroupBy = "variant"
+	GroupByDistro     GroupBy = "distro"
+	SortEarliestFirst Sort    = "earliest"
+	SortLatestFirst   Sort    = "latest"
+)
+
+type GroupBy string
+
+func (gb *GroupBy) validate() error {
```

Comment:
A validation function on a pointer should probably validate that the pointer is not nil.

Reply:
I think it's not correct that these are pointer-recievers. Given that they're strings, I think we just remove some `*` characteers and this all gets better.